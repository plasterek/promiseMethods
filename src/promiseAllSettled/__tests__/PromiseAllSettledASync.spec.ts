import { promiseAllSettledASync } from "../PromiseAllSettledASync.service";
import { IAllSettled } from "../models/IAllSettled.model";

describe("promiseAllSettled function", () => {
  const success: string = "resolved";
  const failure: string = "rejected";
  const resolvedPromise: Promise<string> = Promise.resolve(success);
  const rejectedPromise: Promise<string> = Promise.reject(failure);

  describe("When any promise in given array is resolved", () => {
    it("It returns Promise with array of objects with given promise status and results", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise, resolvedPromise];
      //when
      const results: IAllSettled<string>[] = await promiseAllSettledASync(promiseArray);
      //then
      expect.assertions(1);
      expect(results).toMatchSnapshot();
    });
  });

  describe("When any promise rejects in given array", () => {
    it("It will be returned in promise array with status and reason", async () => {
      //given
      const promiseArray: Promise<string>[] = [resolvedPromise, rejectedPromise];
      //when
      const results: IAllSettled<string>[] = await promiseAllSettledASync(promiseArray);
      //then
      expect.assertions(1);
      expect(results).toMatchSnapshot();
    });
  });

  describe("When given array is empty", () => {
    it("It returns Promise with an empty array", async () => {
      //given
      const promiseArray: any[] = [];
      //when
      const results: IAllSettled<string>[] = await promiseAllSettledASync(promiseArray);
      //then
      expect.assertions(1);
      expect(results).toMatchObject([]);
    });
  });
});
