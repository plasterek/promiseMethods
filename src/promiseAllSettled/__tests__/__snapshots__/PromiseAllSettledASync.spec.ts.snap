// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promiseAllSettled function When any promise in given array is resolved It returns Promise with array of objects with given promise status and results 1`] = `
[
  {
    "status": "fulfilled",
    "value": "resolved",
  },
  {
    "status": "fulfilled",
    "value": "resolved",
  },
]
`;

exports[`promiseAllSettled function When any promise rejects in given array It will be returned in promise array with status and reason 1`] = `
[
  {
    "status": "fulfilled",
    "value": "resolved",
  },
  {
    "status": "rejected",
    "value": "rejected",
  },
]
`;
